### ISSUE TITLE
Jupyter console conficts with prompt-toolkit 3.0.0 

### ISSUE THREAD

**Issue by verga** [jupyter/jupyter_console#193]
After upgrading I get:

    ```ERROR: jupyter-console 6.0.0 has requirement prompt-toolkit<2.1.0,>=2.0.0, 
    but you'll have prompt-toolkit 3.0.0 which is incompatible.
    ```
I restored the previous version  ``pip install -U  prompt-toolkit==2.0.10`` to use jupyter lab

**Comment by cbrnr** [comment]
This has been fixed with the most recent ipython update, but this package needs to be updated as well with the  new requirement (https://github.com/ipython/ipython/blob/master/setup.py#L193).

**Comment by dquist** [comment]
For anyone else running into this conflict, it's because ipython was recently updated to add support for prompt-toolkit v3.x, but jupyter-console only supports v2.x.

You can fix this incompatibility by pinning prompt-toolkit to v2.x in your setup.py.

```python
install_requires: [
    'prompt-toolkit~=2.0.10',
    # ...
],
```

**Comment by cbrnr** [comment]
To clarify, jupyter-console already seems to work with prompt-toolkit >= 3.

**Comment by dquist** [comment]
Yes, it seems you are correct. I only saw there were breaking changes for v3 and assumed some changes would be required, but according to the [upgrade notes](https://python-prompt-toolkit.readthedocs.io/en/master/pages/upgrading/3.0.html) the only breaking changes are related to the async event loop handling - something that's not utilized by this project.

I've created a PR to add support for prompt-toolkit v3.
